tests/jax/components/building/environments_test.py

```diff
@@ -0,0 +1,219 @@
+import functools
+from types import SimpleNamespace
+from typing import Any
+
+import numpy as np
+import pytest
+
+from mava import MAEnvironmentSpec, specs
+from mava.components.jax.building.environments import (
+    EnvironmentSpec,
+    EnvironmentSpecConfig,
+    ExecutorEnvironmentLoop,
+    ExecutorEnvironmentLoopConfig,
+    ParallelExecutorEnvironmentLoop,
+)
+from mava.core_jax import SystemBuilder
+from mava.systems.jax import Builder
+from mava.utils.environments import debugging_utils
+from mava.utils.sort_utils import sort_str_num
+
+
+class AbstractExecutorEnvironmentLoop(ExecutorEnvironmentLoop):
+    """Implement abstract methods to allow testing of class"""
+
+    def on_building_executor_environment_loop(self, builder: SystemBuilder) -> None:
+        """Do nothing. Just implement abstract method"""
+        pass
+
+
+@pytest.fixture
+def test_environment_spec() -> EnvironmentSpec:
+    """Pytest fixture for environment spec"""
+    environment_factory = functools.partial(
+        debugging_utils.make_environment,
+        env_name="simple_spread",
+        action_space="discrete",
+        num_agents=10,
+    )
+
+    config = EnvironmentSpecConfig(environment_factory=environment_factory)
+    test_environment_spec = EnvironmentSpec(config=config)
+    return test_environment_spec
+
+
+@pytest.fixture
+def test_executor_environment_loop() -> ExecutorEnvironmentLoop:
+    """Pytest fixture for executor environment loop"""
+    config = ExecutorEnvironmentLoopConfig(should_update=False)
+    return AbstractExecutorEnvironmentLoop(config=config)
+
+
+@pytest.fixture
+def test_parallel_executor_environment_loop() -> ParallelExecutorEnvironmentLoop:
+    """Pytest fixture for executor environment loop"""
+
+    class ExecutorStatsWrapperClass:
+        def __init__(self, executor_environment_loop: Any) -> None:
+            self.executor_environment_loop = executor_environment_loop
+            self.wrapped = True
+
+    config = ExecutorEnvironmentLoopConfig(
+        should_update=False,
+        executor_stats_wrapper_class=ExecutorStatsWrapperClass,  # type: ignore
```

Comment:
Why the executor_stats_wrapper_class is `ExecutorStatsWrapperClass` while it is mentioned as an object of `EnvironmentLoopStatisticsBase`?